test/PowerShellEditorServices.Test.Shared/Refactoring/Functions/ForeachFunction.ps1

```diff
@@ -0,0 +1,17 @@
+$x = 1..10
+
+function testing_files {
+    param (
+        $x
```

Comment:
Attempting to rename this parameter doesn't work as expected (renames outer, not inner). Needs a new test.

Reply:
Added a test case for this example, it was due to how `GetVariableTopAssignment` worked I have added a statement to skip some steps if the target is within a functiondefinition and within a parameterblock